src/modules/projectSearchContext.js

```diff
@@ -23,13 +23,16 @@ class ProjectSearchContext {
         let inEscape = false;
         let inQuote = false;
         for (let i = 1; i < buffer.length; i++) {
-            const character = buffer[i];
+            const character = buffer.charAt(i);
 
-            if (character === '\\') {
+            if (character === '\\' && inQuote) {
                 inEscape = !inEscape;
             } else if (!inEscape) {
                 if (character === '"') {
-                    inQuote = !inQuote;
+                    if ((i >= 2 && buffer.charAt(i -1) === '\\' && buffer.charAt(i - 2) === '\\') ||
```

Comment:
A line comment might help here. How does the escaping we want to detect here work? How do the escaping cases we want to account for look exactly?

Reply:
code no longer exists, but the new code has new doc